### PR TITLE
feat: Extend `UnitTestGenerator` to support test generation from directories

### LINKED ISSUE #392: Feature Request: Extend `generate_tests` method to support folder paths
The current `generate_tests` method in the `UnitTestGenerator` class only accepts a single file path. This request proposes extending the method to accept a folder path, allowing it to generate tests for all files within the directory.

*Triage needed*

### PR DESCRIPTION
**Pull request by MashyBasker**
# feat: Extend `UnitTestGenerator` to support test generation from directories

## Overview
This pull request extends the functionality of the `UnitTestGenerator` class to support generating unit tests from directories, in addition to individual files. This enhancement allows the generator to recursively process all source files within a specified directory and generate corresponding unit tests.

## Changes
- Key Changes:
  - Added a new method `generate_tests_from_dir` that takes a directory path as input and generates tests for all source files within that directory and its subdirectories.
  - The `generate_tests` method now supports an optional `output_path` parameter to specify the output directory for the generated test files.
  - Refactored the `generate_test_files` method to improve readability and maintainability.
- New Features:
  - The `UnitTestGenerator` can now generate unit tests for all source files within a directory, simplifying the process of creating tests for multiple files.
- Refactoring:
  - Improved the formatting and readability of the output logs during the test generation process.

> ✨ Generated with love by [Kaizen](https://cloudcode.ai) ❤️

<details>
<summary>Original Description</summary>
Changelog:
- Use the existing `generate_tests` method to generate test for each path.
- Use the `pathlib` module to target code present in sub-directories.

Fixes #392 
</details>

### CODE REVIEW

**Review by sauravpanda — APPROVED** [review]
Looks good!

File: kaizen/generator/unit_test.py [hunk 1]
```diff
@@ -33,6 +34,17 @@ def __init__(self):
...
+    def generate_tests_from_dir(
+        self, dir_path: str, output_path: str = None
+    ):
+        """
+        dir_path: (str) - path of the directory containing source files
+        """
+        if output_path:
+            self.output_folder = output_path
+        for file_path in Path(dir_path).rglob('*.*'):
+            self.generate_tests(file_path=str(file_path), output_path=output_path)
```
[the remark this answers is not in the document]
Comment (MashyBasker) [reply kaizen/generator/unit_test.py]: I've made a commit to handle this with a try/exception block.

**Comment by MashyBasker** [comment]
@sauravpanda should I add this function to the docs?

**Comment by sauravpanda** [comment]
Yes that would be great, also if you find other issues with the doc, feel free to create a task!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Pull request merged**